Group.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+#include <iostream>
+#include <vector>
+#include <tuple>
+#include <algorithm>
+#include <memory>
+#include "Team.h"
+
+class Group
+{
+public:
+	std::vector<std::shared_ptr<Team>> m_teams;
+	const double ImportanceOfMatch = 35.0; //FIFA sets different coefficients for different stages of the tournament. For a group round it's 35.
+	const int numberTeams = 4; //the number of teams in the group.
+	Group() = default;
+	~Group() = default;
+	void PlayMatch(std::shared_ptr<Team>& first, std::shared_ptr<Team>& second);
```

Comment:
Out-arguments (arguments that are changed after a function call) are generally discouraged for a number of reasons. Better return the outcome and then apply it to the teams somewhere else.

Reply:
As I see, you only mutate teams by calling CountingStats on them, so you can instead make this method const and just return the stats from it, instead of mutating the team.